src/libpmem2/config.c

```diff
@@ -113,6 +113,12 @@ pmem2_config_validate_length(const struct pmem2_config *cfg,
 		size_t file_len, size_t alignment)
 {
 	ASSERTne(alignment, 0);
+
+	if (file_len == 0) {
+		ERR("file length is equal 0");
+		return PMEM2_E_INVALID_ARG;
```

Comment:
PMEM2_E_INVALID_ARG is meaningless in this case - file size doesn't come from the user, so [s]he has no way of knowing what PMEM2_E_INVALID_ARG refers to.

PMEM2_E_SOURCE_EMPTY?

This also needs to be documented.

Reply:
:+1: